deepspeed/runtime/zero/stage3.py

```diff
@@ -1191,6 +1202,12 @@ def _create_fp16_partitions_with_defragmentation(self):
 
                 see_memory_usage(f"After Flattening param subgroup {i}", force=False)
 
+        # Perhaps it should assert if any params are unpartitioned.
+        if len(unpartitioned_params) > 0 and torch.distributed.get_rank() == 0:
+            logger.warning(
+                f"{len(unpartitioned_params)} unpartitioned params detected:\n" +
+                "\n".join(unpartitioned_params))
+

```

Comment:
I added this as I saw situations where params were missed and weren't partitioned!

I wasn't sure if it should assert or just warn.

the setup is 100 lines earlier.

Reply:
For some reason I can't find the original report, but what you're saying answers that concern.

So let me rework/undo this part. Do we want to optionally log that some params are unpartitioned? Perhaps just commenting out this whole added section?

Or I can revert it complete and revisit it down the road if and when there will be a need for it.

I think I will just revert it for now.

Reply:
ok, reverted.

Reply:
Thanks!